### PR TITLE
feat: add method count wistlist for user

### PR DESCRIPTION
**Pull request by pnlinh**
Hi @lamalamaMark,

This PR added method `countWishlist()` context needle get count wishlist result for the user.

Count all wishlist

```php
$user->countWishlist();
```

Count wishlist by collections

```php
$user->countWishlist(['default', 'Christmas presents']);
$user->countWishlist('default', 'Christmas presents');
```

### CODE REVIEW

File: src/HasWishlists.php [hunk 1]
```diff
@@ -4,9 +4,27 @@
...
 {
+    /**
+     * Count wishlist for user.
+     *
+     * @param  array  $collections
+     * @return int
+     */
+    public function countWishlist(array $collections = []): int
+    {
+        $query = DB::table('wishlist')->where('user_id', Auth::id());
```
Comment (lamalamaMark) [inline src/HasWishlists.php:19]: Please use ```$this->id``` for the user ID for consistency; in this way we can also skip the use of the Auth facade.

File: src/HasWishlists.php [hunk 2]
```diff
@@ -4,9 +4,27 @@
...
 
 trait HasWishlists
 {
+    /**
+     * Count wishlist for user.
+     *
+     * @param  array  $collections
+     * @return int
+     */
+    public function countWishlist(array $collections = []): int
```
Comment (lamalamaMark) [inline src/HasWishlists.php:17]: Can you make this a little more flexibel by making it possible the either add an array (for multiple) or a string (for a single) collection.

**Review by lamalamaMark — COMMENTED** [review]
Many thanks for your contribution 🔥

File: src/HasWishlists.php (hunk 1 above)
Comment (pnlinh) [reply src/HasWishlists.php]: @lamalamaMark yeah, good idea 👍

**Comment by pnlinh** [comment]
@lamalamaMark please review it.

**Pull request closed without merging**